packages/server/src/adapters/ws.ts

```diff
@@ -139,7 +136,9 @@ export function applyWSSHandler<TRouter extends AnyRouter>(
           ctx,
         });
 
-        if (!(result instanceof Subscription)) {
+        // check if returned value is an observable
+        // otherwise just send the value as data
+        if (!(typeof result === 'object' && result && 'subscribe' in result)) {
```

Comment:
is there a better way to detect if something is an observable?

Reply:
It's good enough, let's make it a util that is something like

```ts
function isObservable(x: unknown): asserts X is Observable) {
  //... 
}
```